synapse/events/spamcheck.py

```diff
@@ -270,7 +272,7 @@ def register_callbacks(
 
     async def check_event_for_spam(
         self, event: "synapse.events.EventBase"
-    ) -> Union[Decision, str]:
+    ) -> Union[Codes, str]:
```

Comment:
all the values in `Codes` are `str`s, so I think this is this is equivalent to `-> str`